kinematics_base/include/moveit/kinematics_base/kinematics_base.h

```diff
@@ -56,6 +56,39 @@ class RobotState;
 namespace kinematics
 {
 
+/*
+ * @enum KinematicSearches
+ * @brief Flags for choosing the type discretization applied on the redundancy during a multiple ik solution search
+ */
+namespace KinematicSearches
+{
+  enum KinematicSearch
+  {
+    ONE = 1,
+    ALL_DISCRETIZED ,
+    SOME_DISCRETIZED,
+    RANDOM_SAMPLE,
+    ALL
```

Comment:
Do you actually need to switch between these different flags to have Descartes working? We have done a planner similar to Descartes without needing these additional flags so that's why I am curious as to why you need them.